lib/config/size.go

```diff
@@ -73,3 +75,24 @@ func (s Size) String() string {
 func (Size) ParseDefault(s string) (interface{}, error) {
 	return ParseSize(s)
 }
+
+func (s Size) checkFree(fs fs.Filesystem) error {
+	val := s.BaseValue()
+	if val <= 0 {
+		return nil
+	}
+
+	usage, err := fs.Usage(".")
+	if s.Percentage() {
+		freePct := (float64(usage.Free) / float64(usage.Total)) * 100
+		if err == nil && freePct < val {
+			return fmt.Errorf("insufficient space in %v %v: %f %% < %v", fs.Type(), fs.URI(), freePct, s)
+		}
+	} else {
+		if err == nil && float64(usage.Free) < val {
+			return fmt.Errorf("insufficient space in %v %v: %v < %v", fs.Type(), fs.URI(), usage.Free, s)
+		}
+	}
+
+	return nil
+}
```

Comment:
All of this should live in `func (f *FolderConfiguration) CheckFreeSpace() (err error) {`, and keep using a type, as a type.

Reply:
This code is also needed to get free space of home.

Reply:
Fine, so have a utility function or something, don't just wedge something on a type.

Reply:
Wedging was more or less a random decision, I just thought this makes sense here as it is used in the config package (before it was wedged on `Model`, but it has no connection to that at all). Would it make more sense as a utility function in the fs package?

Reply:
I think so.

Reply:
config fs import cycle...
So either we leave it in config or I move the entire Size thing into the fs package.

Reply:
Make a utility in config.